### ISSUE TITLE
topView may be hidden

### ISSUE THREAD

**Issue by hiworld75018** [runway20/PopoverView#20]
The <code>topView</code> in <code>showAtPoint:inView:withContentView:</code> may lead to a hidden view.

Maybe this works better ?
<code>
NSArray\* windowViews = [[[UIApplication sharedApplication] keyWindow] subviews];
for (int i = windowViews.count - 1; i >= 0 ; i--)
{
    topView = [windowViews objectAtIndex:i];
    if (!topView.isHidden)
        break;
}
</code>

**Comment by devios1337** [comment]
I have to agree with @jallen. Using the immediate superview (or some customizable ancestor) makes a lot more sense. I am using popovers inside a scrollView that I am using to display text field validation warnings. In this case I naturally want the popovers to scroll with the text fields, but by using the topView (or a new UIWindow as you suggest), the popover is outside of the scrollView and remains static on the screen while the underlying view is scrolled, forcing me to hide the validation messages if/when the scrollView is scrolled.

Instead, to solve your problem of the popover being attached to a view with clipping enabled, would it not make more sense for the target view to be overridable? This is a pattern I am employing in my app: if you don't specify the target view, it defaults to self.superview. This lets you specify any parent view (or even a completely different window if so desired) without losing the ability for the popover to be directly attached to the immediate superview and thus contained inside any scrollViews, etc.

I made the following simple changes and things seem to be working great. I probably broke something else, but for my use this is working perfectly (the popover views now scroll appropriately with the views they are attached to):

PopoverView.m
Line 483: topView = view;
Line 525: topViewBounds = topView.frame;

That's all it took, and it works great now. You can still specify what view you want the popover to appear in, and that simply becomes the topView now. No need to actually attach the view to the top-most view.

**Comment by wasauce** [comment]
@devios1337 If you have a pull request we can investigate if there are conflicts and potentially merge
